### PR TITLE
Feature/cart pagination

### PR DESCRIPTION
**Pull request by foxy-methoxy**

### CODE REVIEW

File: src/components/Cart/CartProducts/CartProducts.vue [hunk 1]
```diff
@@ -16,7 +16,30 @@
...
+      />
+    </label>
+    <Arrow
+      @click="nextPage"
+      class="pagination__arrow pagination__arrow--right"
+    />
+  </div>
+  <!--End of pagination-->
+
+  <div v-if="!page.length" class="np-cartProducts__empty">
```
Comment (entrop-) [inline src/components/Cart/CartProducts/CartProducts.vue]: if youre using BEM convention then empty should be a modifier (--empty not __empty)

File: src/components/Cart/CartProducts/CartProducts.vue [hunk 2]
```diff
@@ -64,12 +94,47 @@ export default defineComponent({
     };
 
     watch(store.state.cart.items, updatePrice);
+    const cart = computed(() => store.getters["cart/cartItems"]);
```
Comment (rafalporeba3) [inline src/components/Cart/CartProducts/CartProducts.vue]: :type

File: src/components/Cart/CartProducts/CartProducts.vue [hunk 3]
```diff
@@ -64,12 +94,47 @@ export default defineComponent({
...
+    const page = computed(() => store.getters["cart/getPage"]);
```
Comment (rafalporeba3) [inline src/components/Cart/CartProducts/CartProducts.vue]: :type

File: src/components/Cart/CartProducts/CartProducts.vue [hunk 4]
```diff
@@ -64,12 +94,47 @@ export default defineComponent({
...
+    const pagesCount = computed(() => store.state.cart.pagesCount);
```
Comment (rafalporeba3) [inline src/components/Cart/CartProducts/CartProducts.vue]: :type

File: src/components/Cart/CartProducts/CartProducts.vue [hunk 5]
```diff
@@ -64,12 +94,47 @@ export default defineComponent({
...
+
+    const inputValue: Ref<number> = ref(1);
+    const currentPage: ComputedRef<number> = computed(
+      () => store.state.cart.currentPage
+    );
+    watch(currentPage, () => (inputValue.value = currentPage.value));
+
+    const prevPage = () => {
```
Comment (rafalporeba3) [inline src/components/Cart/CartProducts/CartProducts.vue]: :type

File: src/components/Cart/CartProducts/CartProducts.vue [hunk 6]
```diff
@@ -64,12 +94,47 @@ export default defineComponent({
...
+      store.commit("cart/prevPage");
+      inputValue.value = currentPage.value;
+    };
+    const nextPage = () => {
```
Comment (rafalporeba3) [inline src/components/Cart/CartProducts/CartProducts.vue]: :type

File: src/components/Cart/CartProducts/CartProducts.vue [hunk 7]
```diff
@@ -64,12 +94,47 @@ export default defineComponent({
...
+      store.commit("cart/nextPage");
+      inputValue.value = currentPage.value;
+    };
+    const setPage = (event: ChangeEvent) => {
```
Comment (rafalporeba3) [inline src/components/Cart/CartProducts/CartProducts.vue]: :type

File: src/components/Cart/CartProducts/CartProducts.vue [hunk 8]
```diff
@@ -64,12 +94,47 @@ export default defineComponent({
...
+      const newValue = event.target.value;
```
Comment (rafalporeba3) [inline src/components/Cart/CartProducts/CartProducts.vue]: :type

File: src/store/modules/cart.ts [hunk 9]
```diff
@@ -11,10 +15,35 @@ const getters = {
...
+    state.currentPage < state.pagesCount ? state.currentPage++ : null;
+  },
+  setPage: function(state: CartState, page: number): void {
+    state.currentPage = page;
+  },
+  calculatePagesCount: function(state: CartState) {
+    state.pagesCount = Math.ceil(state.items.length / state.itemsPerPage);
+    state.currentPage > state.pagesCount ? (state.currentPage = 1) : null;
+  },
+  setPages: function(state: CartState) {
```
Comment (rafalporeba3) [inline src/store/modules/cart.ts]: I would change all above to arrow function

File: src/store/modules/cart.ts [hunk 10]
```diff
@@ -11,10 +15,35 @@ const getters = {
...
     return state.priceTotal;
+  },
+  getPage(state: CartState): CartItem[] {
+    return state.paginatedItems[state.currentPage - 1] || [];
   }
 };
 
 const mutations = {
+  prevPage: function(state: CartState): void {
+    state.currentPage > 1 ? state.currentPage-- : null;
```
Comment (rafalporeba3) [inline src/store/modules/cart.ts]: I think this is more readable
```
if(state.currentPage > 1) {
 state.currentPage  -= 1;
}
```

File: src/store/modules/cart.ts [hunk 11]
```diff
@@ -11,10 +15,35 @@ const getters = {
...
+  },
+  nextPage: function(state: CartState): void {
+    state.currentPage < state.pagesCount ? state.currentPage++ : null;
```
Comment (rafalporeba3) [inline src/store/modules/cart.ts]: As above, I would wrap it with `if` statement

File: src/store/modules/cart.ts [hunk 12]
```diff
@@ -11,10 +15,35 @@ const getters = {
...
+  },
+  setPage: function(state: CartState, page: number): void {
+    state.currentPage = page;
+  },
+  calculatePagesCount: function(state: CartState) {
+    state.pagesCount = Math.ceil(state.items.length / state.itemsPerPage);
+    state.currentPage > state.pagesCount ? (state.currentPage = 1) : null;
```
Comment (rafalporeba3) [inline src/store/modules/cart.ts]: I would wrap it with if statement

File: src/store/modules/cart.ts [hunk 13]
```diff
@@ -11,10 +15,35 @@ const getters = {
...
+  },
+  setPages: function(state: CartState) {
+    state.paginatedItems = [];
+    const pageSize: number = state.itemsPerPage;
+    const items: CartItem[] = [...state.items];
+    for (let i = 0; i < items.length; i += pageSize) {
```
Comment (rafalporeba3) [inline src/store/modules/cart.ts:48]: You can also use `.forEach`

Comment (foxy-methoxy) [reply src/store/modules/cart.ts:48]: not really cause I dont want as many iterations as there is items in the array (i += pageSize where pageSize is currently 3 => so for 9 items I do 3 iterations)

**Review by entrop- — APPROVED** [review]

**Pull request merged**